azure/services/networkinterfaces/spec.go

```diff
@@ -125,6 +126,11 @@ func (s *NICSpec) Parameters(existing interface{}) (parameters interface{}, err
 		s.AcceleratedNetworking = &accelNet
 	}
 
+	dnsSettings := network.InterfaceDNSSettings{}
+	if len(s.DNSServers) != 0 {
+		dnsSettings.DNSServers = &s.DNSServers
+	}
+
```

Comment:
@jsturtevant Is it okay to be passing `<nil>` `dnsSettings` in case `s.DNSServers` list is empty?

Reply:
Are you concerned about a nil slice being evaluated as `!=  0`? In fact that's not the case in golang. See:

https://go.dev/play/p/E0ilKGlGiMg

Reply:
Sorry that I confused, I meant to ask https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2447/files#r932527885

Reply:
I think you got your answer :-)  this looks ok to me....